llvm-external-projects/iree-dialects/lib/Transforms/TransformMatchers.cpp

```diff
@@ -1373,13 +1399,18 @@ void transform_ext::makeMatmulMatcher(
     transform_ext::StructuredOpMatcher *&fillCapture,
     transform_ext::StructuredOpMatcher *&trailingCapture,
     transform_ext::MatchedMatmulCaptures &captures, bool mustMatchEntireFunc) {
-  auto &matmul = transform_ext::m_StructuredOp<linalg::MatmulOp>(matcherContext)
-                     // Capture op sizes.
-                     .dim(AllDims(), CaptureDims(captures.matmulOpSizes))
-                     // Capture input/output element types.
-                     .input(0, CaptureElementType(captures.lhsElementType))
-                     .input(1, CaptureElementType(captures.rhsElementType))
-                     .output(0, CaptureElementType(captures.outputElementType));
+  auto &matmul =
+      transform_ext::m_StructuredOp<linalg::GenericOp, linalg::MatmulOp,
```

Comment:
On a second reading, using `GenericOp` here is incorrect. This will match absolutely any generic since we don't have actual constraints below, only captures. This should either avoid the generic or have a predicate for the body being what one expects from the generic.